src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,1810 @@
+        config: FastSpeech2ConformerConfig,
+        attention_heads=4,
+        linear_units=2048,
+        dropout_rate=0.1,
+        attention_dropout_rate=0.0,
+        normalize_before=True,
+        concat_after=False,
+        cnn_module_kernel=31,
+    ):
+        super().__init__()
+
+        # self-attention module definition
+        self.self_attn = FastSpeech2ConformerAttention(attention_heads, config.hidden_size, attention_dropout_rate)
+
+        # feed-forward module definition
+        self.feed_forward = FastSpeech2ConformerMultiLayeredConv1d(
+            config.hidden_size, linear_units, config.positionwise_conv_kernel_size, dropout_rate
+        )
+
+        self.macaron_style = config.use_macaron_style_in_conformer
+        if self.macaron_style:
+            self.feed_forward_macaron = FastSpeech2ConformerMultiLayeredConv1d(
+                config.hidden_size, linear_units, config.positionwise_conv_kernel_size, dropout_rate
+            )
+            self.ff_macaron_layer_norm = nn.LayerNorm(config.hidden_size)
+            self.ff_scale = 0.5
+        else:
+            self.ff_scale = 1.0
+
+        # convolution module definition
+        self.use_cnn_module = config.use_cnn_in_conformer
+        if self.use_cnn_module:
+            self.conv_module = FastSpeech2ConformerConvolutionModule(config, cnn_module_kernel)
+            self.conv_layer_norm = nn.LayerNorm(config.hidden_size)
+            self.final_layer_norm = nn.LayerNorm(config.hidden_size)
+
+        # for the FNN module
+        self.ff_layer_norm = nn.LayerNorm(config.hidden_size)
+
+        # for the MHA module
+        self.self_attn_layer_norm = nn.LayerNorm(config.hidden_size)
+
+        self.dropout = nn.Dropout(dropout_rate)
+        self.size = config.hidden_size
+        self.normalize_before = normalize_before
+        self.concat_after = concat_after
+        if self.concat_after:
+            self.concat_linear = nn.Linear(config.hidden_size + config.hidden_size, config.hidden_size)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        pos_emb: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        output_attentions: Optional[torch.Tensor] = False,
+    ):
+        """
+        Compute encoded features.
+
+        Args:
+            hidden_states (`torch.Tensor` of shape `(batch, time, size)`): Input tensor.
+            pos_emb (`torch.Tensor` of shape `(1, time, size)`): Positional embeddings tensor.
+            attention_mask (`torch.Tensor` of shape `(batch, time)`): Attention mask tensor for the input.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+        Returns:
+            `torch.Tensor`: Output tensor of shape `(batch, time, size)`.
+
+        """
+        # whether to use macaron style
```

Comment:
Seems like there is a lot of `if xxx`, are they all used by the released models? (Meaning if it's only a single pass that is always use, let's remove the complicated logic)

Reply:
Microsoft didn't release their weights, but these if's are all configurable in the library with the implementation this used (espnet). We could remove the logic but it might break converting some that have been trained in espnet